src/m365/spo/commands/page/page-set.ts

```diff
@@ -184,33 +202,56 @@ class SpoPageSetCommand extends SpoCommand {
           data: {
             BannerImageUrl: bannerImageUrl,
             CanvasContent1: canvasContent1,
-            LayoutWebpartsContent: layoutWebpartsContent
+            LayoutWebpartsContent: layoutWebpartsContent,
+            Description: pageDescription
           }
         };
+
         return request.post(requestOptions);
       })
-      .then((): Promise<void> => {
-        if (args.options.promoteAs !== 'Template') {
-          return Promise.resolve();
-        }
+      .then((): Promise<void> => {        
+        let requestOptions: any = {};
 
-        const requestOptions: any = {
-          responseType: 'json',
-          url: `${args.options.webUrl}/_api/SitePages/Pages(${templateListItemId})/SavePageAsDraft`,
-          headers: {
-            'X-RequestDigest': requestDigest,
-            'X-HTTP-Method': 'MERGE',
-            'IF-MATCH': '*',
-            'content-type': 'application/json;odata=nometadata',
-            accept: 'application/json;odata=nometadata'
-          },
-          data: {
-            Title: fileNameWithoutExtension,
-            BannerImageUrl: bannerImageUrl,
-            CanvasContent1: canvasContent1,
-            LayoutWebpartsContent: layoutWebpartsContent
+        if (args.options.promoteAs !== 'Template') {
+          if (pageId) {
```

Comment:
To avoid nesting the code too deep, let's reverse the check and do an early return:

```ts
if (pageId === null) {
  return Promise.resolve();
}
```

Reply:
After applying this change, I came across a logical issue in the code. In the tests `pageId` is coming through as `undefined`. While it passes `if (page)` it fails with `if (page === null)` because `null` and `undefined` are not the same thing. `pageId` is however defined as `number | null` so the question is if it should be `number | null` and the tests should return a proper `null` value or should `pageId` be changed to `number | undefined`?

Reply:
Can indeed be changed to `undefined` or you can change it to:

```
if (!pageId) {
  return Promise.resolve();
}
```

As the `pageId` will always contain a value when adding a page. When setting it might not, but that means the page doesn't exist and it would already fail before this check.

Reply:
I was hesitant to use `!pageId` at first, because a `0` would give incorrect outcome, but I don't think `0` is a realistic value for the id so we could indeed simplify to `!pageId` rather than checking the exact type.

Reply:
IDs all start at 1, so we are good that way.

Reply:
In that case, it's all ready for review and we'll get it in asap